text/0098-store-multiple-replay-segments-in-a-single-blob.md

```diff
@@ -0,0 +1,91 @@
+- Start Date: 2023-05-24
+- RFC Type: decision
+- RFC PR: https://github.com/getsentry/rfcs/pull/98
+- RFC Status: draft
+
+# Summary
+
+Recording data is sent in segments. Each segment is written to its own file. Writing files is the most expensive component of our Google Cloud Storage usage. It is also the most expensive component, in terms of time, in our processing pipeline. By merging many segment files together into a single file we can minimize our costs and maximize our processing pipeline's throughput.
+
+# Motivation
+
+1. Minimize costs.
+2. Improve throughput.
+3. Enable new features in a cost-effective manner.
+
+# Background
+
+This document exists to inform all relevant stakeholders of our proposal and seek feedback prior to implementation.
+
+# Supporting Data
+
+Google Cloud Storage lists the costs for writing and storing data as two separate categories. Writing a file costs $0.005 per 1000 files. Storing that file costs $0.02 per gigabyte. For the average file this works out to: $0.000000012 for storage and $0.00000005 for the write.
+
+In practical terms, this means 75% of our spend is allocated to writing new files.
+
+# Proposal
+
+First, a new table called "recording_byte_range" with the following structure is created:
+
+| replay_id | segment_id | path     | start | stop  |
```

Comment:
i wonder if we could store this on the corresponding replay event for the segment?

Reply:
We could.  We would need to merge the replay-event and recording-event payloads first.  But this means that we wouldn't be able to store segment parts separately according to the `Extensions` section.

Reply:
Am I correctly understanding that one blob would store the segments for an arbitrary amount of different replays?

Reply:
I wouldn't say arbitrary but any number of segments from different replays and different projects.

Reply:
And the current procedure is storing one blob per segment?